### ISSUE TITLE
S905X4, S905W2 and S905Y2

### ISSUE THREAD

**Issue by educabox** [devmfc/amlogic-bootscripts-Armbian#4]
I'm curious if the existing tutorials are suitable for the S905X4, S905W2 and S905Y2 processors. Additionally, I would like to know if it is possible to use the following DTBs to run an Armbian image:

Experimental: S905X4 (Meson SC2)
Experimental: S905W2 (Meson S4)
Experimental: S905Y4 (S4 Meson)

Any information or guidance on this would be greatly appreciated. Thank you in advance!

**Comment by devmfc** [comment]
No, Armbian does NOT support these SoC's on an mainline Linux version. 

AFAIK, only the Khadas Vim1S (and maybe some Radxa SBC?) image, based on a vendor kernel will support a S905Y4 SoC. But the vendor kernel is not compatible with mainline Linux DTB's of my images, so using the DTB's of my images is a no go.

**Comment by educabox** [comment]
We're interested in knowing if it's feasible to enable support for the S905X4 and S905Y2 SoCs on Armbian. Could you provide an update on any ongoing development of compatible .dtbs for these newer processor models? Given their increasing adoption in new TVs and devices, we appreciate your partnership and innovative efforts in advancing support for these processors.

Thank you in advance for your insights.

Best regards!

**Comment by devmfc** [comment]
I'm sorry, I think there must be some misunderstanding. 

The scripts in this repo are just a hack/workaround to use Armbian on TVBoxes without the need to have a generic bootloader provided by some guys in the Armbian community. I really don't understand why they still want to do that, since there is AFAICT no reason to do that. They only increase the chance of (soft) bricking a box and decrease the chance they have the right firmware version for the memory in these boxes. Likely resulting in crashes, slower memory speeds or inability to boot at all.

Armbian doesn't officially support TVBoxes. And I don't develop anything for Armbian, and I'm not going to. 

Furthermore, you are talking about a partnership? Sorry, I really like the goals you have with EducaBox, but this is just a side project/hobby for me. I am  also a professional developer (not as a kernel developer), and I am for hire, but I guess probably not for a fee you can afford. So there is no partnership. at all..?

To make a mainline kernel work for the newer SoC's, it is really not that simple of just putting together a simple DTB. You will also need to modify/add some drivers to support the new SoC's. And all that without access to a reference manual of these SoC's.

So if you want to use these SoC's you can use my Debian/Ubuntu images. But the support of these newer SoC's are labeled "Experimantal" for good reason. So if I can give you some advice: stick with the older SoC's! They are supported out of the box by mainline Linux. It will take years before the newer SoC's get proper support (if at all).

If you only want to use Armbian (Why? What is wrong with standard Debian/Unbuntu?), you should really be asking them to support these SoC's.

Best regards!

**Comment by educabox** [comment]
Good evening,

Thank you very much for your feedback! First of all, I apologize for the way I spoke to you and for not introducing myself earlier. My name is Lucas, and I am a Computer Science teacher in Brazil. I teach high school students at a technical school.

As part of my work at the school, we have a partnership with our regulatory body (Receita Federal) regarding seized contraband and pirated products sold in Brazil, including various TV Boxes that illegally stream signals from paid broadcasters. We reconfigure these devices into desktop computers so they can be donated to charities, public schools, NGOs, and government departments.

In this context, the creation of the Educabox project on GitHub aims to share the knowledge we've gained from reconfiguring these TV Boxes. This will enable other teachers, technicians, and professionals across the country to convert similar devices, thereby promoting access to information in underserved communities and public schools lacking equipped computer labs and internet access.

Currently, we have successfully reconfigured around 15 different devices. However, we are facing challenges with the new S905X4 and S905Y2 processors provided by the regulatory body. Our goal was to transform these TV Boxes into desktop PCs with a graphical interface for donation purposes, but unfortunately, we have not found adequate documentation. I attempted to use your Debian/Ubuntu images, but I did not succeed with these Boxes, possibly due to missing a step in the tutorial.

Once again, thank you for your feedback, and I fully understand your project on GitHub. I apologize for any inconvenience caused. If you could provide further guidance on how to run your Debian/Ubuntu images with a graphical interface on these devices, I would greatly appreciate your assistance.

Best regards!

**Comment by devmfc** [comment]
Hi,

Yes, I already kind of knew what you are doing. And I admire your goals very much. 

Unfortunately there is no working mainline HDMI driver for these SOC's. So also with my images you can not use these boxes as a graphical desktop. And I do not have any real knowledge regarding the Linux HDMI/Graphical subsystems so, it is not very likely I will be able to fix that for you. 

Until then, all S904X4/S905W2/S905Y4 SOC's can only be used as servers when you want to use mainline kernels.

If you really want to uses them for desktop use, I think your only chance is to try/build some Khadas Fenix images with Amlogic vendor kernels. Or a modified Armbian image for the Khadas Vim1S/VIM4 which also uses vendor kernels.

I'm sorry I cant help you.

Best regards

**Comment by afgp87** [comment]
> I'm curious if the existing tutorials are suitable for the S905X4, S905W2 and S905Y2 processors. Additionally, I would like to know if it is possible to use the following DTBs to run an Armbian image:
> 
> Experimental: S905X4 (Meson SC2)
> Experimental: S905W2 (Meson S4)
> Experimental: S905Y4 (S4 Meson)
> 
> Any information or guidance on this would be greatly appreciated. Thank you in advance!

Have you being available to installing any linux variant on s905x4 I wonder if is any updates

**Comment by devmfc** [comment]
https://github.com/devmfc/debian-on-amlogic

**Comment by freezedi** [comment]
> https://github.com/devmfc/debian-on-amlogic
https://github.com/devmfc/debian-on-amlogic
how we can downloaded that image? cant find the direct link on repo with img. I did you start on it

**Comment by projetotvbox** [comment]
Hi! We are the TVBox Project from IFSP Salto, partners in device repurposing and digital inclusion! We've been following Educabox's work closely and it has been a great inspiration for our own documentation efforts.

Regarding the S905X4 (Meson SC2): it still doesn't have mainline kernel support, which means Armbian doesn't officially support it either. The existing tutorials are not directly applicable to this SoC.

That said, there is a workaround we've had success with:

1. Build/flash an Armbian image targeting S905X3
2. Replace the kernel with the one from **devmfc's debian-on-amlogic** project: https://github.com/devmfc/debian-on-amlogic
3. You may also need to update the `extraargs` line in `armbianEnv.txt` to:
```
extrargs=earlycon=meson,0xfe07a000 console=ttyS0,921600n8 rootflags=data=writeback rw no_console_suspend consoleblank=0 fsck.fix=yes fsck.repair=yes net.ifnames=0 watchdog.stop_on_reboot=0 pd_ignore_unused clk_ignore_unused rootdelay=5
```

It's not plug and play, but we've managed to get it booting. As for S905W2 and S905Y2, we don't have hands-on experience with those, so we'd rather not comment.

As a side note: we're currently working on documenting the full repurposing process for the **HTV H8**, which uses the S905X4 SoC. We'll publish the repository soon — hopefully it will be useful for anyone tackling this chip.

Hope this helps!

**Comment by educabox** [comment]
Hi @projetotvbox,

Thank you very much for sharing this information and for taking the time to document your experience with the S905X4.

We would like to clarify that we had already successfully completed this process in 2024. However, since we have not updated the EducaBox repository in recent months, some of our documentation and materials have become outdated.

We really appreciate your contribution and the technical details you shared. It is valuable to see other institutions working on similar challenges and developing new approaches for repurposing these TV Boxes.

We are very happy to see new techniques and methods being developed to extend the useful life of these devices and make them suitable for educational and technological applications. This type of collaboration is exactly what helps the project move forward.

Thank you again for the contribution and for sharing your experience with us.

Best regards,

EducaBox Team
